src/rust/ide/lib/enso-protocol/src/language_server/types.rs

```diff
@@ -314,6 +314,13 @@ pub struct TextRange {
     pub end: Position
 }
 
+impl TextRange {
+    /// Returns true if range does not contain any characters.
+    pub fn is_empty(&self) -> bool {
+        self.start == self.end
```

Comment:
Is `end` < `start` is allowed? If so, is the intended meaning an empty or a reversed range?

Reply:
actually `end` < `start` does not work at all in Language Server (we get the error in response). And I'm not sure how we want to interpret this.

Reply:
Maybe then we should go with the behaviour for [`std::ops::Range`](https://doc.rust-lang.org/std/ops/struct.Range.html) and say if `end <= start` the range is empty.

Reply:
Ou, this turned out to not being used anyway. I'll remove it then.